bundles/org.eclipse.passage.lic.api/src/org/eclipse/passage/lic/internal/api/conditions/mining/MinedConditions.java

```diff
@@ -33,6 +33,6 @@
  */
 public interface MinedConditions extends Service<StringServiceId> {
 
-	Collection<Condition> all();
+	Collection<Condition> all() throws ConditionMiningException;
```

Comment:
The contract is not yet clear, please descrbe real cases. Typically we have a folder that contains several license files. If the first file cannot be decrypted (we will throw Exception) the second with valid license will be ignored.

Reply:
May be we need an abstraction of `ConditionLode` (as we have miner :) ) to represent the potential source of `LicensePack` (encrypted license file) and the mining may fail for particular "lode". 

Another variant is to call the miner with callback - it may be even more flexible.

Reply:
The exception is more to handle infrastructural misbehaviour like operating system denials or net communication failures. I suppose it's on the actual implementation,  but we definitely need this possibility to exist. 

The case you've pointed might be implemented like that: exception is thrown if something is wrong with the directory itself or file system throws IOException of attepmt to operate on the directory's content.  Invalid license should not cause the exception firing, if the logic of the miner suppose successes during the further work.